ui/app/services/path-help.js

```diff
@@ -42,89 +44,130 @@ export default Service.extend({
+    }
+
+    //use paths to dynamically create our openapi help url
+    //if we have a brand new model
+    return this.getPaths(apiPath, backend, itemType)
+      .then(pathInfo => {
         const adapterFactory = owner.factoryFor(`adapter:${modelType}`);
         //if we have an adapter already use that, otherwise create one
         if (!adapterFactory) {
           debug(`Creating new adapter for ${modelType}`);
-          const adapter = this.getNewAdapter(paths, itemType);
+          const adapter = this.getNewAdapter(pathInfo, itemType);
           owner.register(`adapter:${modelType}`, adapter);
         }
+        let path, paths;
         //if we have an item we want the create info for that itemType
-        let path;
-        if (itemType) {
-          const createPath = paths.create.find(path => path.path.includes(itemType));
-          path = createPath.path;
-          path = path.slice(0, path.indexOf('{') - 1) + '/example';
-        } else {
-          //we need the mount config
-          path = paths.configPath[0].path;
+        paths = itemType ? this.filterPathsByItemType(pathInfo, itemType) : pathInfo.paths;
+        const createPath = paths.find(path => path.operations.includes('post') && path.action !== 'Delete');
+        path = createPath.path;
+        path = path.includes('{') ? path.slice(0, path.indexOf('{') - 1) + '/example' : path;
+        if (!path) {
+          return reject();
         }
-        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true`;
+
+        helpUrl = `/v1/${apiPath}${path.slice(1)}?help=true` || newModel.proto().getHelpUrl(backend);
+        pathInfo.paths = paths;
+        newModel = newModel.extend({ paths: pathInfo });
         return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
+      })
+      .catch(err => {
+        console.error(err);
       });
-    }
   },
 
-  reducePaths(paths, currentPath) {
+  reducePathsByPathName(pathInfo, currentPath) {
     const pathName = currentPath[0];
-    const pathInfo = currentPath[1];
+    const pathDetails = currentPath[1];
+    const displayAttrs = pathDetails['x-vault-displayAttrs'];
 
-    //config is a get/post endpoint that doesn't take route params
-    //and isn't also a list endpoint and has an Action of Configure
-    if (
-      pathInfo.post &&
-      pathInfo.get &&
-      (pathInfo['x-vault-displayAttrs'] && pathInfo['x-vault-displayAttrs'].action === 'Configure')
-    ) {
-      paths.configPath.push({ path: pathName });
-      return paths; //config path should only be config path
+    if (!displayAttrs) {
+      return pathInfo;
     }
 
-    //list endpoints all have { name: "list" } in their get parameters
-    if (pathInfo.get && pathInfo.get.parameters && pathInfo.get.parameters[0].name === 'list') {
-      paths.list.push({ path: pathName });
+    if (pathName.includes('{')) {
+      //we need to know if there are url params
+      pathName.split('{')[1].split('}')[0];
```

Comment:
The result of this isn't being saved anywhere, I _think_ this can go, or if its meant to be doing something then we need to fix it up so it does what it's meant to do. P.S. oh I think it might be related to line 140 further down?